tests/test_kms.py

```diff
@@ -42,3 +44,118 @@ def test_key_rotation(self):
 
         resources = p.run()
         self.assertEqual(len(resources), 2)
+
+
+    @functional
+    def test_kms_remove_matched(self):
+        session_factory = self.replay_flight_data('test_kms_remove_matched')
+        client = session_factory().client('kms')
+        key_id = client.create_key()['KeyMetadata']['KeyId']
+        self.addCleanup(client.schedule_key_deletion, KeyId=key_id, PendingWindowInDays=7)
+
+        client.put_key_policy(
+            KeyId=key_id,
+            PolicyName='default',
+            Policy=json.dumps({
+                "Version": "2008-10-17",
+                "Statement": [
+                    {
+                      "Sid": "DefaultRoot",
+                      "Effect": "Allow",
+                      "Principal": {
+                        "AWS": "arn:aws:iam::123456789012:root"
+                      },
+                      "Action": "kms:*",
+                      "Resource": "*"
+                    },
+                    {
+                        "Sid": "SpecificAllow",
+                        "Effect": "Allow",
+                        "Principal": {
+                            "AWS": "arn:aws:iam::123456789012:root"
+                        },
+                        "Action": [
+                            "kms:*"
+                        ]
+                    },
+                    {
+                        "Sid": "Public",
+                        "Effect": "Allow",
+                        "Principal": "*",
+                        "Action": [
+                            "kms:Put*"
+                        ]
+                    }
+                ]
+            })
+        )
+
+        p = self.load_policy({
+            'name': 'kms-rm-matched',
+            'resource': 'kms-key',
+            'filters': [
+                {'KeyId': key_id},
+                {'type': 'cross-account',
+                 'whitelist': ["123456789012"]}],
+            'actions': [
+                {'type': 'remove-statements',
+                 'statement_ids': 'matched'}]
+            },
+            session_factory=session_factory)
+        
+        resources = p.run()
+        self.assertEqual([r['KeyId'] for r in resources], [key_id])
+        #time.sleep(60) # takes time before new policy reflected
```

Comment:
any sleeps for functional tests should be behind a conditional for if the test is in recording mode. ie.

```
if self.recording:
    time.sleep(10)
```